### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by Redmar-van-den-Berg**

### CODE REVIEW

File: vtools/cli.py [hunk 1]
```diff
@@ -31,11 +31,26 @@
               help="Sample(s) in positive-vcf to consider. "
                    "May be called multiple times",
               required=True)
-def evaluate_cli(call_vcf, positive_vcf, call_samples, positive_samples):
+@click.option("-s", "--stats", type=click.Path(writable=True),
+              help="Path to output stats json file", default='-')
```
Comment (rhpvorderman) [inline vtools/cli.py:35]: I am glad '-' works. using `sys.stdout` is maybe more specific, but this maybe some click related stuff.

File: vtools/cli.py [hunk 2]
```diff
@@ -31,11 +31,26 @@
...
+                 discordant):
     c_vcf = VCF(call_vcf, gts012=True)
     p_vcf = VCF(positive_vcf, gts012=True)
-    evaluated = site_concordancy(c_vcf, p_vcf, call_samples, positive_samples)
-    print(json.dumps(evaluated))
+    st, disc = site_concordancy(c_vcf, p_vcf, call_samples,
+                                positive_samples)
+    # Write the stats json file
+    with click.open_file(stats, 'w') as fout:
+        print(json.dumps(st), file=fout)
```
Comment (rhpvorderman) [inline vtools/cli.py:47]: I believe there is also a fout.write() function that accomplishes the same.

File: vtools/cli.py [hunk 3]
```diff
@@ -31,11 +31,26 @@
...
+
+    # If specified, write the discordant variants
+    if discordant:
+        with click.open_file(discordant, 'w') as fout:
+            for record in disc:
+                print(record, file=fout, end='')
```
Comment (rhpvorderman) [inline vtools/cli.py:53]: fout.writeline could also work instead of print.

**Review by rhpvorderman — APPROVED** [review]
Some alternatives to using print. The code looks good!

File: vtools/cli.py (hunk 3 above)
Comment (Redmar-van-den-Berg) [reply vtools/cli.py:53]: The problem with using fout.write here is that `record` is not a string, but a `cyvcf2.cyvcf2.Variant`. So we would need to do something like `fout.write(str(record))`, which is not a lot nicer than using the current `print` statement. It's just unfortunate that converting a `record` to string adds a newline character, so we also have to specify `end=''` to the print statement.

Comment (rhpvorderman) [reply vtools/cli.py:53]: `print()` always adds a newline. `write()` should not.  `writeline()` also adds a newline.
If `str(record)` adds a newline already, then `fout.write(str(record))` should work as expected and is less strange (by convention) than using `print` to write to a file.

Comment (rhpvorderman) [reply vtools/cli.py:53]: But since it works, and it the code is totally readable I approved it :+1:

**Pull request merged**